### ISSUE TITLE
[BUG] App freezes when dismissing android browser

### ISSUE THREAD

**Issue by ms-emp** [Badisi/auth-js#41]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Library version

latest

### Description

When running on Android and the user dismisses the login window (by clicking the X button), the app becomes unresponsive.

It happens more with the angular library.

It looks like there are two browser windows open, if you click the device back button we can see another window fading down and then the app is responsive again.

### Steps to reproduce

1. clone the repo
2. run the `start:android:demo-ngx-auth` script
3. click Login
4. dismiss the browser window
5. try to click Login again
6. if it works, dismiss the browser and repeat
7. at some point the app will become unresponsive

If you minimize the app and then bring it back to the foreground it will work again.

### Environment

- **OS Name**:
- **System Model Name**:
- **npm**:
- **Node.js**:

**Comment by Badisi** [comment]
This issue seems related to `@capacitor/browser`.

By default this is the plugin used on mobile to open a web view and navigate to the identity provider.
This web view is opened in a separate **native** window, which means it is out of our control.
Same applies when closing the window in step 4: we are closing it manually (not programmatically).
So it all happens in the native/capacitor context (not in our app context).

I have opened an issue in the Capacitor repo : https://github.com/ionic-team/capacitor-plugins/issues/2045
with detailed explanations and a video.

**Comment by Badisi** [comment]
Worth mentioning here that I'm also planning on releasing a cordova/capacitor plugin to better follow Apple's recommendations.

Currently **@capacitor/browser** is using `SFSafariViewController` instead of `SFAuthenticationSession (iOS11)` and `ASWebAuthenticationSession (iOS12+)`. Not only it does not follow the security guideline but it also not allow SSO to work, as SFSafariViewController does not share cookies outside the app.

**Comment by ms-emp** [comment]
> This issue seems related to `@capacitor/browser`.
> 
> By default this is the plugin used on mobile to open a web view and navigate to the identity provider. This web view is opened in a separate **native** window, which means it is out of our control. Same applies when closing the window in step 4: we are closing it manually (not programmatically). So it all happens in the native/capacitor context (not in our app context).
> 
> I have opened an issue in the Capacitor repo : [ionic-team/capacitor-plugins#2045](https://github.com/ionic-team/capacitor-plugins/issues/2045) with detailed explanations and a video.

But when I try to use the browser plugin directly (for example: `Browser.open({ url: 'http://capacitorjs.com/' })`) it never happens?!

**Comment by Badisi** [comment]
@ms-emp, I traced the issue down to something like:
```ts
await Browser.addListener(
  'browserFinished',
  (): void => void Browser.close())
);
await Browser.open({ url: 'http://capacitorjs.com/' });
```
Sounds like the issue is when we are trying to close the browser when it is actually closed or being closed.
Fun fact: the [doc](https://capacitorjs.com/docs/apis/browser#close) says it should be a no-op for Android.

Can you please confirm this scenario is also failing on your side ?

**Comment by ms-emp** [comment]
@Badisi 

You are right, `Browser.close()` is causing the issue, so probably we shouldn't call it on Android.

**Comment by ms-emp** [comment]
@Badisi 

When do you plan to release the latest version?

**Comment by Badisi** [comment]
Hopefully this week-end, or next Monday _(still have a few issues to fix)_.

**Comment by Badisi** [comment]
:tada: Released in [@badisi/auth-js@1.0.0](https://github.com/Badisi/auth-js/releases/tag/@badisi/auth-js@1.0.0) :package::rocket:

**Comment by Badisi** [comment]
:tada: Released in [@badisi/ngx-auth@3.0.0](https://github.com/Badisi/auth-js/releases/tag/@badisi/ngx-auth@3.0.0) :package::rocket:
